internal/distro/distro_test.go

```diff
@@ -16,3 +18,81 @@ func TestDistro_Manifest(t *testing.T) {
 		distroregistry.NewDefault(),
 	)
 }
+
+var (
+	v1manifests = []string{
+		`{}`,
+		`
+{
+	"sources": {
+		"org.osbuild.files": {
+			"urls": {}
+		}
+	},
+	"pipeline": {
+		"build": {
+			"pipeline": {
+				"stages": []
+			},
+			"runner": "org.osbuild.rhel84"
+		},
+		"stages": [],
+		"assembler": {
+			"name": "org.osbuild.qemu",
+			"options": {}
+		}
+	}
+}`,
+	}
+
+	v2manifests = []string{
+		`{"version": "2"}`,
+		`
+{
+	"version": "2",
+	"pipelines": [
+		{
+			"name": "build",
+			"runner": "org.osbuild.rhel84",
+			"stages": []
+		}
+	],
+	"sources": {
+		"org.osbuild.curl": {
+			"items": {}
+		}
+	}
+}`,
+	}
+)
+
+func TestDistro_Version(t *testing.T) {
+	assert := assert.New(t)
+	expectedVersion := "1"
+	for idx, rawManifest := range v1manifests {
+		manifest := distro.Manifest(rawManifest)
+		detectedVersion, err := manifest.Version()
+		assert.NoError(err, "Could not detect Manifest version for %d: %v", idx, err)
+		assert.Equal(expectedVersion, detectedVersion, "in manifest %d", idx)
```

Comment:
IMO in this case it would be better to use require.NoError (fails fast) instead of assert. Other than than the assertion functions are the same.

Reply:
FTR in general with Go I prefer the `require` package over the `assert` one but I see we've used both of them.

Reply:
I prefer letting it run through the rest of the tests even if one fails (where possible) since they're mostly independent and it's good to know if one or all of them are failing.  But it's not that important here.  Changed it to require.